components/sst_importer/src/sst_importer.rs

```diff
@@ -530,8 +534,19 @@ impl<E: KvEngine> SSTWriter<E> {
     }
 
     // move file from temp to save.
-    fn save(mut import_path: ImportPath) -> Result<()> {
+    fn save(mut import_path: ImportPath, key_manager: Option<&Arc<DataKeyManager>>) -> Result<()> {
```

Comment:
can we make it `save(&mut self, ...)`?

Reply:
better use `save(mut self, ...)` than `save(&mut self, ...)` as it is not supposed to be called multiple times

Reply:
save is call after `SSTWriter::finish(self)`, so at this time, self is gone because of the need to move `SSTWriter`.

Reply:
it is called *inside* `SSTWriter::finish`, not after. the actual reason `self` is "gone" is because several fields are moved out making the struct incomplete.